Xamarin.Essentials/Screenshot/Screenshot.macos.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using ImageIO;
+using MobileCoreServices;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Screenshot
+    {
+        internal static bool PlatformIsCaptureSupported => true;
+
+        static Task<ScreenshotResult> PlatformCaptureAsync()
+        {
+            var image = ApplicationServices.GetScreenshot();
+            var result = new ScreenshotResult(image);
+            return Task.FromResult(result);
+        }
+    }
+
+    public partial class ScreenshotResult
+    {
+        readonly CGImage cgImage;
+
+        internal ScreenshotResult(CGImage image)
+        {
+            cgImage = image;
+            Width = (int)cgImage.Width;
+            Height = (int)cgImage.Height;
+        }
+
+        internal Task<Stream> PlatformOpenReadAsync(ScreenshotFormat format)
+        {
+            var utType = UTType.JPEG;
+            switch (format)
```

Comment:
A switch with a single case can be rewritten as an if

Reply:
I would have used `if` but I used the style from iOS:
https://github.com/xamarin/Essentials/blob/0dba844e8085f7cb36d62115b005620d541e0fbd/Xamarin.Essentials/Screenshot/Screenshot.ios.tvos.cs#L35

Reply:
The difference here is that they actually use two cases, the two swtich statements are not similar. Yours is a overly complicated if block, they have an if/else block

Reply:
Thats correct. In my reading style I prefer to use the full statement, because the default knows nobody. And wie would be safe if there will be a new type. Would be this OK?

```csharp
NSString utType;
switch (format)
{
    case ScreenshotFormat.Png:
        utType = UTType.PNG;
        break;
    case ScreenshotFormat.Jpeg:
        utType = UTType.JPEG;
        break;
    default:
        throw new NotImplementedException("The ScreenshotFormat is not supported");
}
```